lib/hyrax/specs/shared_specs/indexers.rb

```diff
@@ -1,11 +1,12 @@
 # frozen_string_literal: true
 
 RSpec.shared_examples 'a Hyrax::Resource indexer' do
-  subject(:indexer) { indexer_class.new(resource: resource) }
-  let(:ids)         { ['id1', 'id2'] }
-  let(:resource)    { Hyrax::Resource.new(alternate_ids: ids) }
+  subject(:indexer)  { indexer_class.new(resource: the_resource) }
+  let(:ids)          { ['id1', 'id2'] }
+  let(:the_resource) { defined?(resource) ? resource : Hyrax::Resource.new }
 
   describe '#to_solr' do
+    before { the_resource.alternate_ids = ids }
```

Comment:
Makes sure the alternate_ids are set whether the resource is created in the shared spec or by the caller.

NOTE: Similar use of before blocks in other shared specs to be sure the resource has the values set that are being tested.